EIPS/eip-5187.md

```diff
@@ -0,0 +1,102 @@
+
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```
+pragma solidity ^0.8.0;
+
+interface IRental {
+    /**
+     * @notice This emits when user rent NFT
+     * @param id The id of the current token
+     * @param user The address to rent the NFT usage rights
+     * @param amount The amount of usage rights
+     * @param expires The specified period of time to rent
+     **/
+    event Rented(
+        uint256 indexed id,
+        address indexed user,
+        uint256 amount,
+        uint256 expires
+    );
+
+    /**
+     * @notice This emits when the NFT owner takes back the usage rights from the tenant (the `user`) 
+     * @param id The id of the current token
+     * @param user The address to rent the NFT's usage rights
+     * @param amount Amount of usage rights
+     **/
+    event TakeBack(uint256 indexed id, address indexed user, uint256 amount);
+
+    /**
+     * @notice Function to rent out usage rights
+     * @param from The address to approve
+     * @param to The address to rent the NFT usage rights
+     * @param id The id of the current token
+     * @param amount The amount of usage rights
+     * @param expires The specified period of time to rent
+     **/
+    function safeRent(
+        address from,
+        address to,
+        uint256 id,
+        uint256 amount,
+        uint256 expires
+    ) external;
+
+    /**
+     * @notice Function to take back usage rights after the end of the tenancy
+     * @param user The address to rent the NFT's usage rights
+     * @param tokenId The id of the current token
+     * @param amount The amount of usage rights
+     **/
+    function takeBack(
+        address user,
+        uint256 tokenId,
+        uint256 amount
+    ) external;
+
+    /**
+    * @notice Return the NFT to the NFT owner’s address
+    **/
+    function ownerOf(uint256 id) external view returns (address);
+
+    /**
+    * @notice Return the total supply amount of the current token
+    **/
+    function totalSupply(uint256 id) external view returns (uint256);
+}
+
+
+```
+## Rationale
+There are two main benefits to creating this proposal. One is to make it possible to create NFTs with detachable usage rights, of which the transfer of usage rights is separated from the transfer of ownership. The NFT owner can execute the safeRent function to lease out the usage rights to other users and the takeBack function to retrieve the usage rights after the lease expiration.
+
+## Backwards Compatibility
+As mentioned at the beginning, this is an extension of EIP-1155. Therefore, it is fully compatible with EIP-1155.
+
+## Security Considerations
+There are no security considerations directly related to the implementation of this standard.
```

Comment:
```suggestion
Needs discussion.
```
Personally, I prefer explicitly stating that potential security issues may have not cropped up while this is a draft.